segmentstore/server/src/test/java/io/pravega/segmentstore/server/DataRecoveryTestUtils.java

```diff
@@ -0,0 +1,188 @@
+import io.pravega.segmentstore.contracts.tables.IteratorArgs;
+import io.pravega.segmentstore.contracts.tables.IteratorItem;
+import io.pravega.segmentstore.contracts.tables.TableKey;
+import io.pravega.segmentstore.server.containers.DebugStreamSegmentContainer;
+import io.pravega.segmentstore.server.tables.ContainerTableExtension;
+import io.pravega.segmentstore.storage.SegmentHandle;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.shared.segment.SegmentToContainerMapper;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+import static io.pravega.shared.NameUtils.getMetadataSegmentName;
+
+/**
+ * Utility methods for data recovery tests.
+ */
+@Slf4j
+public class DataRecoveryTestUtils {
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+    private static final ScheduledExecutorService EXECUTOR_SERVICE = createExecutorService(10);
+
+    /**
+     * Lists all segments from a given long term storage.
+     * @param tier2             Long term storage.
+     * @param containerCount    Total number of segment containers.
+     * @return                  A map of lists containing segments by container Ids.
+     * @throws                  IOException in case of exception during the execution.
+     */
+    public static Map<Integer, List<SegmentProperties>> listAllSegments(Storage tier2, int containerCount) throws IOException {
+        SegmentToContainerMapper segToConMapper = new SegmentToContainerMapper(containerCount);
+        Map<Integer, List<SegmentProperties>> segmentToContainers = new HashMap<Integer, List<SegmentProperties>>();
+        log.info("Generating container files with the segments they own...");
+        Iterator<SegmentProperties> it = tier2.listSegments();
+        if (it == null) {
+            return segmentToContainers;
+        }
+        // Iterate through all segments. Put each one of them in its respective list.
+        while (it.hasNext()) {
+            SegmentProperties curr = it.next();
+            int containerId = segToConMapper.getContainerId(curr.getName());
+            List<SegmentProperties> segmentsList = segmentToContainers.get(containerId);
+            if (segmentsList == null) {
+                segmentsList = new ArrayList<>();
+                segmentsList.add(curr);
+                segmentToContainers.put(containerId, segmentsList);
+            } else {
+                segmentToContainers.get(containerId).add(curr);
+            }
+        }
+        return segmentToContainers;
+    }
+
+    public static ScheduledExecutorService createExecutorService(int threadPoolSize) {
```

Comment:
This is not needed. Looks copy pasted from `ExecutorServiceHelpers`

Reply:
Removed.